extensions/resource-deployment/src/interfaces.ts

```diff
@@ -18,17 +18,36 @@ export interface ResourceType {
 	icon: { light: string; dark: string } | string;
 	options: ResourceTypeOption[];
 	providers: DeploymentProvider[];
-	agreement?: AgreementInfo;
+	agreement?: AgreementInfo[];
 	displayIndex?: number;
 	okButtonText?: OkButtonTextValue[];
 	getOkButtonText(selectedOptions: { option: string, value: string }[]): string | undefined;
 	getProvider(selectedOptions: { option: string, value: string }[]): DeploymentProvider | undefined;
+	getAgreementInfo(selectedOptions: { option: string, value: string }[]): AgreementInfo | undefined;
+	getHelpText(selectedOption: { option: string, value: string }[]): string | undefined;
+	tags?: string[];
+}
+
+export interface ResourceSubType {
+	/**
+	 * The name should match the name in Resource Type
+	 */
+	name: string;
+	/**
+	 * The option name should have a matching name in ResourceType.options
+	 */
+	options: ResourceTypeOption[];
 	tags?: string[];
+	provider: DeploymentProvider;
+	okButtonText?: OkButtonTextValue;
+	helpText?: string;
+	agreement?: AgreementInfo;
```

Comment:
You changed ResourceType agreement to be an array but this is still a single one?

Reply:
Yeah, because sub resource type can provide only one provider, agreement, okButtonText

Reply:
Hm, ok. But I would say then that agreement in ResourceType should be agreements instead. What types do you have that need to support multiple agreements? It might make more sense to have agreement be a part of the options then for those cases